src/java.base/share/classes/jdk/internal/classfile/instruction/ExceptionCatch.java

```diff
@@ -0,0 +1,89 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.instruction;
+
+import java.util.Optional;
+
+import jdk.internal.classfile.CodeElement;
+import jdk.internal.classfile.CodeModel;
+import jdk.internal.classfile.constantpool.ClassEntry;
+import jdk.internal.classfile.Label;
+import jdk.internal.classfile.PseudoInstruction;
+import jdk.internal.classfile.impl.AbstractPseudoInstruction;
+
+/**
+ * A pseudo-instruction modeling an entry in the exception table of a code
+ * attribute.  Entries in the exception table model catch and finally blocks.
+ * Delivered as a {@link CodeElement} when traversing the contents
+ * of a {@link CodeModel}.
+ *
+ * @see PseudoInstruction
+ */
+public sealed interface ExceptionCatch extends PseudoInstruction
+        permits AbstractPseudoInstruction.ExceptionCatchImpl {
+    /**
+     * {@return the handler for the exception}
+     */
+    Label handler();
+
+    /**
+     * {@return the beginning of the instruction range for the guarded instructions}
+     */
+    Label tryStart();
+
+    /**
+     * {@return the end of the instruction range for the guarded instructions}
+     */
+    Label tryEnd();
+
+    /**
+     * {@return the type of the exception to catch, or empty if this handler is
+     * unconditional}
+     */
+    Optional<ClassEntry> catchType();
+
+    /**
+     * {@return an exception table pseudo-instruction}
+     * @param handler the handler for the exception
+     * @param tryStart the beginning of the instruction range for the gaurded instructions
+     * @param tryEnd the end of the instruction range for the gaurded instructions
+     * @param catchTypeEntry the type of exception to catch, or empty if this
+     *                       handler is unconditional
+     */
+    static ExceptionCatch of(Label handler, Label tryStart, Label tryEnd,
+                             Optional<ClassEntry> catchTypeEntry) {
+        return new AbstractPseudoInstruction.ExceptionCatchImpl(handler, tryStart, tryEnd, catchTypeEntry);
```

Comment:
```suggestion
        return new AbstractPseudoInstruction.ExceptionCatchImpl(handler, tryStart, tryEnd, catchTypeEntry.orElse(null));
```
Then you only need one constructor for `ExceptionCatchImpl` and you don't need to disambiguate a call using a cast for a final argument that is `null` .